From: "Jiajin Cui" <cuijiajin@uniontech.com>
#Subject: [PATCH] winex11.drv: Fix window state error when WS_MAXIMIZE style.
Message-Id: <202105281009363914811@uniontech.com>
Date: Fri, 28 May 2021 10:10:36 +0800

From ac29f87099fe29b40fbe9fd4a9d393da21bcf155 Mon Sep 17 00:00:00 2001
From: Jiajin Cui <cuijiajin@uniontech.com>
Date: Fri, 28 May 2021 09:39:29 +0800
Subject: [PATCH] winex11.drv: Fix window state error when WS_MAXIMIZE style.

net_state must remove NET_WM_STATE_FULLSCREEN and NET_WM_STATE_MAXIMIZED Atoms when window have WS_MAXIMIZE style.

Signed-off-by: Jiajin Cui <cuijiajin@uniontech.com>
---
 dlls/winex11.drv/window.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/winex11.drv/window.c b/dlls/winex11.drv/window.c
index 11111111111..11111111111 100644
--- a/dlls/winex11.drv/window.c
+++ b/dlls/winex11.drv/window.c
@@ -1089,7 +1089,7 @@ void update_net_wm_states( struct x11drv_win_data *data )
     }
 
     if (style & WS_MINIMIZE)
-        new_state |= data->net_wm_state & ((1 << NET_WM_STATE_FULLSCREEN)|(1 << NET_WM_STATE_MAXIMIZED));
+        new_state |= data->net_wm_state & ~((1 << NET_WM_STATE_FULLSCREEN)|(1 << NET_WM_STATE_MAXIMIZED));
     if (fullscreen)
     {
         if ((style & WS_MAXIMIZE) && (style & WS_CAPTION) == WS_CAPTION)

-- 
2.37.1

